fitbenchmarking/fitbenchmarking
# Cost function linting fixes

#### Description of Work

Fixes #766 

There are still some warnings:
```
************* Module fitbenchmarking.cost_func.poisson_cost_func
cost_func/poisson_cost_func.py:33:4: W0235: Useless super delegation in method '__init__' (useless-super-delegation)
************* Module fitbenchmarking.cost_func.tests.test_cost_func
cost_func/tests/test_cost_func.py:256:8: W0201: Attribute 'options' defined outside __init__ (attribute-defined-outside-init)
```
that may or may not be easy to fix...?

#### Testing Instructions

1. run flake8, pylint on code
2. check changes don't break anything
3.

Function: Does the change do what it's supposed to?

Tests: Does it pass? Is there adequate coverage for new code?

Style: Is the coding style consistent? Is anything overly confusing?

Documentation: Is there a suitable change to documentation for this change?
